API/Endpoints/ClassroomEndpoints.cs

```diff
@@ -51,10 +52,14 @@ public static WebApplication UseClassroomEndpoints(this WebApplication app)
 
         }).RequireAuthorization(nameof(Roles.Instructor)).WithRequestValidation<ClassroomSessionDto>();
 
-        classroomV2.MapGet("/{classroomId:int}", async Task<Ok<GetClassroomResponseDto>> (int classroomId, ClaimsPrincipal principal, IClassroomService service) =>
+        classroomV2.MapGet("/{classroomId:int}", async Task<Results<Ok<GetClassroomResponseDto>, BadRequest<string>>> (int classroomId, ClaimsPrincipal principal, IClassroomService service) =>
         {
             var userId = principal.Claims.First(c => c.Type == ClaimTypes.UserData).Value;
             var result = await service.GetClassroomById(classroomId);
+            if(result.IsFailed)
+            {
```

Comment:
Use CreateBadRequest.CreateValidationProblemDetails, look at session endpoints for inspiration.